### ISSUE TITLE
Cop idea: prefer argument forwarding syntax

### ISSUE THREAD

**Issue by pocke** [rubocop-hq/rubocop#7549]
## Is your feature request related to a problem? Please describe.

Since Ruby 2.7, we will be able to use `...` to forward arguments to another method.

```ruby
# 2.6
def foo(*args, &block)
  bar(*args, &block)
end

# 2.7
def foo(...)
  bar(...)
end
```

I think the `...` syntax has two advantages.

First, we can simply avoid keyword parameter warnings with `...` syntax.
For example

```ruby
# with Ruby 2.7

def bar(*args, a: 1, &block)
  p [args, a]
end

# It displays the warning.
def foo(*args, &block)
  bar(*args, &block)
end
foo 1, 2, a: 1

# It displays no warning, but it is a bit of long.
undef foo
def foo(*args, **kwargs, &block)
  bar(*args, **kwargs, &block)
end
foo 1, 2, a: 1

# No warning, and short!
undef foo
def foo(...)
  bar(...)
end

foo 1, 2, a: 1
```

Second, we can avoid missing `&block` forwarding with `...` syntax.
I guess many people forget passing `&block` to the forwarded method unexpectedly.
But we can avoid the problem with `...` syntax.

## Describe the solution you'd like

Add an offense for the following code.

```ruby
# We can replace the parameter with `...`.
# It can avoid the warning.
# It does not change the behavior at least Ruby 2.7.
# The behavior will be changed in the future.
def foo(*args, &block)
  bar(*args, &block)
end

# We can replace the parameter with `...`.
# The behaviors are completely the same in all Rubies.
# So it is cosmetic change only.
def foo(*args, **kwargs, &block)
  bar(*args, **kwargs, &block)
end

# We can replace the parameter with `...`, but it will change the behavior.
# Because `...` forwards block also.
def foo(*args)
  bar(*args)
end
```

The last example changes the method behavior actually, so probably we should add `Strict` option to disable/enable replacing the last case.

## Describe alternatives you've considered

nothing

## Additional context

nothing

**Comment by bbatsov** [comment]
That's a great idea and we should totally implement such a cop!
